specification/programmableconnectivity/Azure.ProgrammableConnectivity/apis/location.tsp

```diff
@@ -0,0 +1,87 @@
+import "./common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+
+namespace Azure.ProgrammableConnectivity;
+
+/// Interfaces
+
+interface LocationInterface {
+  @doc("Verifies whether a device is within a specified location area, defined as an accuracy (radius) around a point, specified by longitude and latitude.")
+  verify is Operations.ResourceAction<
+    LocationEndpoint,
+    LocationVerifyRequest,
+    LocationVerifyResponse
+  >;
+}
+
+/// Endpoints
+
+@resource("DeviceLocation")
+@doc("Static endpoint to access the Device Location API family.")
+model LocationEndpoint {
+  @key
+  @doc("Static endpoint")
+  @visibility("read")
+  location: LocationEnum;
+}
+
+/* Needed to add this enum because typespec was creating an x-ms-enum without a name
+   which was causing a failure during linting. */
+@doc("Location static endpoint enum")
+enum LocationEnum {
+  @doc("Static endpoint")
+  location: "Location",
+}
+
+/// Request models
+
+@doc("Request to verify Location")
+model LocationVerifyRequest {
+  @doc("Network to query for this device, or device information to enable network routing.")
+  networkIdentifier: NetworkIdentifier;
+
+  @doc("Latitude of location to be verified")
+  @minValue(-90)
+  @maxValue(90)
+  latitude: float32;
+
+  @doc("Longitude of location to be verified")
+  @minValue(-180)
+  @maxValue(180)
+  longitude: float32;
+
+  @doc("Accuracy expected for location verification in kilometers")
+  @minValue(2)
+  @maxValue(100)
+  accuracy: int32;
+
+  @doc("The device to find the location for.")
+  locationDevice: LocationDevice;
+}
+
+/// Response models
+
+@doc("Response verifying location")
+model LocationVerifyResponse {
+  @doc("True if the location is in the specified area, False otherwise")
+  verificationResult: boolean;
+}
+
+/// Common models
+@doc("Device information needed by operator to provide location information. Include exactly one of these properties to identify your device.")
+model LocationDevice {
+  @doc("External identifier or network access identifier of the device")
+  networkAccessIdentifier?: string;
+
+  ...PhoneNumberModel;
+
+  @doc("IPv4 address and port of the device")
+  ipv4Address?: Ipv4Address;
```

Comment:
I don't want to use the spread operator because I want it to be obvious that these are grouped together and keeps the "levels" seperate. If we look at the examples (obvisulsy a user wouldn't use all of these at once.). Without spread:
      "locationDevice": {
        "networkAccessIdentifier": "122345@domain.com",
        "phoneNumber": "+447000000000",
        "ipv4Address": {
          "ipv4": "12.12.12.12",
          "port": 2442
        },
        "ipv6Address": {
          "ipv6": "3001:0da8:75a3:0000:0000:8a2e:0370:7334",
          "port": 1643
        }
      }
and the actual call:

      "locationDevice": {
        "ipv4Address": {
          "ipv4": "12.12.12.12",
          "port": 2442
        },
      }

With spread
      "locationDevice": {
        "networkAccessIdentifier": "122345@domain.com",
        "phoneNumber": "+447000000000",
         "ipv4": "12.12.12.12",
         "port": 2442
         "ipv6": "3001:0da8:75a3:0000:0000:8a2e:0370:7334",
         "port": 1643
      }
and the actual call:

      "locationDevice": {
         "ipv4": "12.12.12.12",
         "port": 2442
      }
      
     It makes it look like "port" is on the same level as "phoneNumber", which it is not. And I think it makes it less obvious what needs to be included when you can't just say you need to include 1 of the top level objects and they need to include whatever they need. Maybe you have a different intuition here?

Reply:
I agree with your dislike of that second structure. My suggestion included an extra model to provide that extra level of hierarchy - when you spread my suggested `Ipv4AddressModel` into the RequestBody, you should get something that looks like your first structure, rather than your second.

Reply:
I don't know whether I really buy that that is a better structure, seems like an unnecessary layer of abstraction. But I don't think it has real downsides either so will add it

Reply:
It's just an aesthetic suggestion, it's up to you!

Reply:
Have changed this and the external id for consistency